src/app/client/src/app/modules/learn/services/course-consumption/course-consumption.service.ts

```diff
@@ -153,4 +154,12 @@ getAllOpenBatches(contents) {
   get getCoursePagePreviousUrl()  {
     return this.coursePagePreviousUrl;
   }
+
+  isCourseMentor (courseHierarchy) {
+    const isTrackable = _.lowerCase(_.get(courseHierarchy, 'trackable.enabled')) === 'yes';
+    const courseCreator = this.permissionService.checkRolesPermissions(['CONTENT_CREATOR'])
+    && this.userService.userid === _.get(courseHierarchy, 'createdBy');
+    const courseMentor =  courseCreator ? true : this.permissionService.checkRolesPermissions(['COURSE_MENTOR']) ? true : false;
```

Comment:
If the login user & creator are same(with COURSE_CREATOR or COURSE_MENTOR role), don't look for COURSE_MENTOR role check at all. Allow him to createBatch & viewDashboard.